Networking/WebSockets/BuiltInWebSocket.cc

```diff
@@ -252,7 +254,22 @@ namespace litecore::websocket {
                                             "Missing TLS client cert in C4Replicator config"_sl));
                 return false;
             }
-            if ( slice keyData = auth[kC4ReplicatorAuthClientCertKey].asData(); keyData ) {
+            if ( bool isExternal = auth[kC4ReplicatorAuthClientCertKeyIsExternal].asBool(); isExternal ) {
+#ifdef COUCHBASE_ENTERPRISE
+                if ( !parameters().externalKey ) {
+                    closeWithError(c4error_make(LiteCoreDomain, kC4ErrorInvalidParameter,
+                                                "Missing externalKey when 'clientCertKeyIsExternal' is true"_sl));
+                    return false;
+                }
+                Retained<crypto::Cert> cert = new crypto::Cert(certData);
+                _tlsContext->setIdentity(
+                        new crypto::Identity(cert, const_cast<C4KeyPair*>(parameters().externalKey)->getPrivateKey()));
```

Comment:
It's cleaner to use `make_retained` than `new`. (I know there's old code that uses `new`, but that's before we had `make_retained`.)

Reply:
will do